Add elevator subsystem and ElevatorMove command

mvoodarla opened the pull request:
Still need to start integrating into drive code to be able to move to distinct positions.

mvoodarla committed:
Add elevator subsystem and ElevatorMove command

mvoodarla committed:
Add new constructor to ElevatorMove to move to a specific position

Co-authored-by: Tejas Narayanan <tejasn100@gmail.com>

tnarayanan committed:
Add ElevatorTicksTest to test the encoder length of the elevator

ElevatorTicksTest measures the encoder length of the elevator. This
number can be used to calculate encoder ticks per inch of elevator lift.

AG6GR reviewed (changes requested):
Some minor comments/fixes.

mvoodarla committed:
Modify elevator code to correctly move to specific encoder positions

Fixed code so that the elevator will move to the encoder positions
specified in Elevator.java

tnarayanan committed:
Allow for elevator use with either manual control or discrete positions

Currently, the elevator can be moved with either discrete positions or
manual control, but not both. The goal is to rely mostly on discrete
positions, but the elevator should be able to be controlled manually at
any point.

tnarayanan committed:
Update elevator test code to use either wheel or buttons for control

Controlling the elevator using the wheel and the buttons at the same
time does not currently work. This commit allows the tester to switch
between the two control methods using the autonomous and tele-op modes.
Furthermore, the elevator speed is now slowed when going down to avoid a
fast drop.

iamef committed:
Fix typo in ElevatorMove.java

It used to say “elevator mvoement”
Now it says “elevator movement”

iamef committed:
Modified resetEncoders to stop when encoders are actually reset

Used a while loop and function hasResetOccurred() and set a timer
Set encoderResetComplete to false just in case since it never was
Made Elevator move SLOW_POWER public and final
Add comments

iamef committed:
Make body of if statement on separate line in isFinished() of ElevatorMove

iamef committed:
Remove unused imports ElevatorTicksTest.java

hi1210 committed:
Add Elevator Status to shuffleboard

Added an extra section to the execute portion of elevator move to check
if the elevator is moving an expected number of ticks per cycle (added
in Robot Map, though currently untested) and will display the status
on the SmartDashboard.
This commit is untested.

tnarayanan committed:
Make elevator code stable

Wait for talon reset code caused problems, this has been commented out
to make the elevator code stable

Angelaq621 committed:
Change elevator value in shuffleboard

Sets the value in shuffleboard based on the level of the elevator. It is set to
an integer for each level and decimals in between values. The command did not
run so the code is not tested and further testing must be done.

Angelaq621 committed:
Resolve merge

mvoodarla committed:
Add new commands and changes to Elevator.java

mvoodarla committed:
Merge branch 'master' into ElevatorSubsystem

mvoodarla committed:
Fix errors in Robot, RobotMap, and OI

mvoodarla committed:
Add elevator controls to TeleopDrive

mvoodarla committed:
Fix indentation in Robot.java
